### ISSUE TITLE
(loader_componentlist.ini) WebSocket does not Work on Windows 10 Tablet

### ISSUE THREAD

**Issue by klues** [asterics/AsTeRICS-Ergo#84]
On a Windows 10 Tablet AsTeRICS-Ergo does not work, because the envControl.acs-Model cannot be deployed.
Error is: Failure on start of WebSocket-Plugin

**Comment by klues** [comment]
@stefaniehoechtl 
please insert details about your tablet and your windows-version

@deinhofer 
please insert the details about the websocket-error on model deployment (if you have more).

**Comment by stefaniehoechtl** [comment]
Microsoft Windows 10 Home (Version 10.0.15063 Build 15063)
Acer Switch Alpha 12 (SA5-271)

**Comment by deinhofer** [comment]
It seems that the Websocket plugin could not be started, although this should just register at an internal websocket service. At startup everything was fine, but when the model envControl.js was supposed to be deployed, the start of the Websocket plugin failed.

Here is a log file snippet:
[log_websocket_error.txt](https://github.com/asterics/AsTeRICS-Ergo/files/1374472/log_websocket_error.txt)
important lines:
```
FEIN: resourceName=envControl.acs, componentTypeId=null, runtimeComponentInstanceId=null, Resource URI <file:/C:/AsTeRICS%20Ergo/app/models/envControl.acs>
Okt 02, 2017 7:04:27 PM eu.asterics.mw.are.AsapiSupport deployModel
...
WARNUNG: eu.asterics.mw.are.AsapiSupport.deployModel: Failed to deploy model -> 
Deployment Error: Couldn't start bundle asterics.WebSocket
```

**Comment by klues** [comment]
@deinhofer I found out how to attach a file -> just drop it to to comment box (or click on the link on bottom of the comment box to select a file). I did this for your last comment in order to keep this thread better readable.

@stefaniehoechtl  @all 
I installed Windows 10 Home on a virtual machine and tried to start AsTeRICS-Ergo there. It worked - no WebSocket error! Windows Version is: Windows 10 Home, Version 10.0.15063, so the same version as on the tablet.

**Comment by klues** [comment]
we just found the error by chance :)
error:
1) The folder `C:\Users\<User>\AppData\Local\AsTeRICS Ergo\app\profile` contains the file `loader_componentlist.ini` which defines which components are needed/used by AsTeRICS-Ergo
2) These files are not deleted on uninstall
3) @stefaniehoechtl had installed a previous version of AsTeRICS-Ergo, which did not use WebSockets -> `loader_componentlist.ini` did not contain the Websocket-Component
4) After installing the new Version of AsTeRICS-Ergo the file `loader_componentlist.ini` was still the old one, not containing the Websocket-Component, although it is needed now
5) On deploying the model `envControl.acs` the error occurs, because the component WebSocket is not found, because it is not in `loader_componentlist.ini`

Possible Solutions:
1) Delete the file `loader_componentlist.ini` on uninstalling AsTeRICS-Ergo
2) Refresh the file `loader_componentlist.ini` if a component is not found on deploying a model in ARE
@deinhofer can you implement (2)?

**Comment by klues** [comment]
@stefaniehoechtl in order to fix this problem on your computer you just can delete the file:
`C:\Users\<YourUsername>\AppData\Local\AsTeRICS Ergo\app\profile\loader_componentlist.ini`
after that your AsTeRICS-Ergo should work.

**Comment by klues** [comment]
with https://github.com/asterics/AsTeRICS-Ergo/pull/90 now `loader_componentlist.ini` is deleted on uninstall. so for the future the problem should be fixed, however it nevertheless would be good to also do https://github.com/asterics/AsTeRICS/issues/172

**Comment by klues** [comment]
@stefaniehoechtl: I just created a new release: https://github.com/asterics/AsTeRICS-Ergo/releases/tag/v3.3.0
if you install it, uninstall it, and reinstall it your problem should be solved and AsTeRICS-Ergo should work on your tablet :)

**Comment by stefaniehoechtl** [comment]
@klues 
I did as you described and now it works on both my and Peter's computer. :)
However, on my computer I had to set up the control of the FS20 devices every time after restarting the computer until I deleted the file C:\Users\<YourUsername>\AppData\Local\AsTeRICS Ergo\app\profile\loader_componentlist.ini - now it works without any problems. On Peter's computer I did not have to delete this file, however, I had to install, uninstall & reinstall AsTeRICS Ergo.
Is there a way to make it work properly without needing to delete this file & without needing to install, uninstall & reinstall AsTeRICS Ergo?
PS: Controlling the environment using the smartphone works without any problems.

**Comment by klues** [comment]
hmm, strange...
so what exactly was your problem with FS20?
(1) you had to reinstall FS20 every time or
(2) you had to re-add a learned FS20 device?
if (2): learned IrTrans commands remained and you did not have to re-learn them?

normal, on a computer where AsTeRICS-Ergo was never installed none of the problems with the file `loader_componentlist.ini` should occur again. And with release 3.3. the file should be deleted on uninstall, so I do not understand why you had to delete it manually. However, with https://github.com/asterics/AsTeRICS/issues/172 the problem should also be fixed without the need of delete or uninstall anything.

**Comment by stefaniehoechtl** [comment]
After restarting the computer, all devices, which I had already set up, were there. The control of the devices using the IrTrans worked without any problem. The control of the devices using the FS20 system did not work any more. I had to select "New Device" - "Table / Floor-Lamp" - I was asked whether I had already bought the FS20 - I had to reinstall the FS20 system. Afterwards I was able to set up the control of the Table / Floor-Lamp again and the control worked. This happened after every restart of the computer, until I deleted the loader_componentlist. Then it worked fine.

Tomorrow I might have the opportunity to try on a computer, where AsTeRICS Ergo has never been installed. ;)
